extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/GeneratedStaticResourcesProcessor.java

```diff
@@ -0,0 +1,137 @@
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import io.quarkus.bootstrap.classloading.ClassPathElement;
+import io.quarkus.bootstrap.classloading.QuarkusClassLoader;
+import io.quarkus.deployment.IsDevelopment;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.GeneratedResourceBuildItem;
+import io.quarkus.deployment.builditem.HotDeploymentWatchedFileBuildItem;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+import io.quarkus.paths.FilteredPathTree;
+import io.quarkus.paths.PathFilter;
+import io.quarkus.paths.PathVisitor;
+import io.quarkus.vertx.http.deployment.devmode.NotFoundPageDisplayableEndpointBuildItem;
+import io.quarkus.vertx.http.deployment.spi.AdditionalStaticResourceBuildItem;
+import io.quarkus.vertx.http.deployment.spi.GeneratedStaticResourceBuildItem;
+import io.quarkus.vertx.http.runtime.GeneratedStaticResourcesRecorder;
+import io.quarkus.vertx.http.runtime.RouteConstants;
+import io.quarkus.vertx.http.runtime.handlers.DevClasspathStaticHandler;
+
+/**
+ * {@link GeneratedStaticResourcesProcessor} is responsible for dealing {@link GeneratedStaticResourceBuildItem}
+ * creating a {@link DevClasspathStaticHandler} to handle all static resources
+ * generated from extensions through {@link GeneratedStaticResourceBuildItem} build item.
+ */
+public class GeneratedStaticResourcesProcessor {
+
+    private static final int ROUTE_ORDER = RouteConstants.ROUTE_ORDER_BEFORE_DEFAULT + 60;
+    private static final String META_INF_GENERATED_RESOURCES = "META-INF/generated-resources";
+
+    @BuildStep
+    public void produceResources(List<GeneratedStaticResourceBuildItem> generatedStaticResources,
+            BuildProducer<GeneratedResourceBuildItem> generatedResourceBuildItem,
+            BuildProducer<NativeImageResourceBuildItem> nativeImageResourcesProducer,
+            LaunchModeBuildItem launchModeBuildItem,
+            BuildProducer<AdditionalStaticResourceBuildItem> additionalStaticResourcesProducer) {
+
+        for (GeneratedStaticResourceBuildItem generatedStaticResource : generatedStaticResources) {
+            String generatedStaticResourceLocation = buildGeneratedStaticResourceLocation(generatedStaticResource);
+
+            generatedResourceBuildItem.produce(
+                    new GeneratedResourceBuildItem(generatedStaticResourceLocation,
+                            generatedStaticResource.getContent(), false));
+            // We can't use the vert.x StaticHandler for tests as it doesn't support 'quarkus' protocol
```

Comment:
We could also fix it to support it.

Reply:
Yes, we can always resolve to it if that gets fixed without changing the GeneratedStaticResourceBuildItem api